crates/typst/src/model/heading.rs

```diff
@@ -203,13 +220,33 @@ impl Show for Packed<HeadingElem> {
     fn show(&self, engine: &mut Engine, styles: StyleChain) -> SourceResult<Content> {
         let span = self.span();
         let mut realized = self.body().clone();
+
+        let hanging_indent = self.hanging_indent(styles);
+
+        let mut indent = match self.hanging_indent(styles) {
+            Smart::Custom(length) => length.resolve(styles),
+            Smart::Auto => Abs::zero(),
+        };
+
         if let Some(numbering) = (**self).numbering(styles).as_ref() {
-            realized = Counter::of(HeadingElem::elem())
+            let numbering = Counter::of(HeadingElem::elem())
                 .display_at_loc(engine, self.location().unwrap(), styles, numbering)?
                 .spanned(span)
-                + HElem::new(Em::new(0.3).into()).with_weak(true).pack()
-                + realized;
+                + HElem::new(Em::new(0.3).into()).with_weak(true).pack();
```

Comment:
I think the HElem shouldn't be part of the numbering variable. For the measurement, it is not taken into account anyway due to `weak: true`.